pmd-core/src/main/java/net/sourceforge/pmd/internal/LanguageServiceBase.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * BSD-style license; for more info see http://pmd.sourceforge.net/license.html
+ */
+
+package net.sourceforge.pmd.internal;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.Iterator;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.ServiceConfigurationError;
+import java.util.ServiceLoader;
+
+import net.sourceforge.pmd.annotation.InternalApi;
+
+@InternalApi
+public abstract class LanguageServiceBase<T> {
```

Comment:
I think this could have just been suppressed, if we're merging PMD and CPD languages anyway